File: report/src/main/java/com/hartwig/actin/report/pdf/ReportWriterFactory.java
package com.hartwig.actin.report.pdf;

import org.jetbrains.annotations.NotNull;

public final class ReportWriterFactory {

    private ReportWriterFactory() {
    }

    @NotNull
    public static ReportWriter createProductionReportWriter(@NotNull String outputDirectory) {
        return new ReportWriter(true, outputDirectory);
    }

    @NotNull
    public static ReportWriter createInMemoryReportWriter() {
        return new ReportWriter(false, null);
    }
}

File: report/src/main/kotlin/com/hartwig/actin/report/pdf/ReportWriterFactory.kt
package com.hartwig.actin.report.pdf

object ReportWriterFactory {

    fun createProductionReportWriter(outputDirectory: String, labels: ReportLabels = ReportLabels.load()): ReportWriter {
        return ReportWriter(true, outputDirectory, labels)
    }

    fun createInMemoryReportWriter(labels: ReportLabels = ReportLabels.load()): ReportWriter {
        return ReportWriter(false, null, labels)
    }
}